### PR TITLE
TC: Expand test cases of Libc Math

### PR DESCRIPTION
**Pull request by lokeshbv**
Add new test cases for log10, log10f, log10l.
Expand test cases for exp2, exp2f, exp2l, j0, j0f, j1, j1f, jn, jnf,
y0, y0f, y1, y1f, yn, ynf, rint, rintf, rintl.

Signed-off-by: Lokesh B V <lokesh.bv@partner.samsung.com>

### CODE REVIEW

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 1]
```diff
@@ -2100,6 +2105,78 @@ static void tc_libc_math_log2l(void)
...
+ * @return               :void
+ */
+static void tc_libc_math_log10l(void)
+{
+	const long double in_val[] = { 1.0, INFINITY, NAN, -1.0, 10.0 };
+	const long double sol_val[] = { ZERO, INFINITY, NAN, NAN, 1.0 };
+	long double ret_val[SIZE(sol_val, long double)];
+	int log10l_idx;
+
+	for (log10l_idx = 0; log10l_idx < SIZE(in_val, long double); log10l_idx++) {
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: The ret_val is defined with sol_val size, but here is in_val. Is it right?

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c [hunk 2]
```diff
@@ -3271,16 +3349,18 @@ static void tc_libc_math_y1f(void)
  */
 static void tc_libc_math_yn(void)
 {
-	const double in_val[] = { ZERO, VAL1, -VAL1, VAL2, -VAL2, INFINITY, -INFINITY, NAN };
-	const double sol_val[] = { -INFINITY, ZERO, NAN, ZERO, NAN, ZERO, NAN, NAN };
-	double ret_val[SIZE(sol_val, double)];
+	const double in_val[] = { ZERO, VAL1, -VAL1, VAL2, -VAL2, INFINITY, -INFINITY, NAN, 0x1p302 };
+	const double sol_val[][7] = { { INFINITY, -INFINITY, -INFINITY, -INFINITY, -INFINITY, -INFINITY, -INFINITY }, { ZERO, ZERO, -ZERO, -ZERO, ZERO, ZERO, -ZERO }, { NAN, NAN, NAN, NAN, NAN, NAN, NAN }, { ZERO, ZERO, -ZERO, -ZERO, ZERO, ZERO, -ZERO }, { NAN, NAN, NAN, NAN, NAN, NAN, NAN }, { -ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO }, { NAN, NAN, NAN, NAN, NAN, NAN, NAN }, { NAN, NAN, NAN, NAN, NAN, NAN, NAN }, { -ZERO, -ZERO, ZERO, ZERO, -ZERO, -ZERO, ZERO } };
+	double ret_val[SIZE(sol_val, double)][7];
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: same as previous

Comment (lokeshbv) [reply apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: Here, it should be in_val size, I will update it.

File: apps/examples/testcase/le_tc/kernel/tc_libc_math.c (hunk 1 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/kernel/tc_libc_math.c]: As both in_val and sol_val size are same, I used it.
actually same changes are there in few other tc also. I will update them also.

**Pull request merged**